.github/workflows/test_coverage.yml

```diff
@@ -287,6 +290,27 @@ jobs:
           cd scripts/code_coverage_report/generate_code_coverage_report
           swift run CoverageReportGenerator --presubmit "firebase/firebase-ios-sdk" --head-commit "${GITHUB_SHA}" --token $(gcloud auth print-identity-token) --xcresult-dir "/Users/runner/test/codecoverage" --log-link "https://github.com/firebase/firebase-ios-sdk/actions/runs/${GITHUB_RUN_ID}" --pull-request-num ${{github.event.pull_request.number}} --base-commit "$base_commit"
           fi
+      - name: Incremental Code Coverage
+        if: github.event.pull_request.merged != true && github.event.action != 'closed'
+        env:
+          base_commit: ${{ needs.check.outputs.base_commit }}
+        run: |
+          # Get Head commit of the branch, instead of a merge commit created by actions/checkout.
+          GITHUB_SHA=$(cat $GITHUB_EVENT_PATH | jq -r .pull_request.head.sha)
+          # Get a JSON of `git diff` from the base commit.
+          git diff -U0 ${base_commit} "${GITHUB_SHA}" | scripts/code_coverage_report/git_diff_to_json.sh > scripts/code_coverage_report/generate_code_coverage_report/git_diff.json
+          cat "scripts/code_coverage_report/generate_code_coverage_report/git_diff.json"
+          if [ -d "${{steps.download.outputs.download-path}}" ]; then
+            # Create an uncovered_file_lines.json including code lines not covered by tests.
+            cd scripts/code_coverage_report/generate_code_coverage_report
+            swift build
+            .build/debug/IncrementalCoverageReportGenerator --changed-files "git_diff.json" --file-archive-root-path "${GITHUB_WORKSPACE}" --xcresult-dir "${{steps.download.outputs.download-path}}" --uncovered-line-file-json "uncovered_file_lines.json"
```

Comment:
Instead of doing `swift build` then going to the debug binary, you can call it with `swift run`:

```
swift run IncrementalCoverageReportGenerator --changed-files .....
```